File: Problems/NepsAcademy/e45.py



n = int(input())

valores = list()
somaLinha = list()
somaColuna = list()
resp = -1

for i in range(n):
    valores.append(list(map(int, input().split())))
    
    
# soma linha
for i in range(n):
    soma = 0
    for j in range(n):
        soma += valores[i][j]
    
    somaLinha.append(soma)
    

# soma coluna
for i in range(n):
    soma = 0
    for j in range(n):
        soma += valores[j][i]
    
    somaColuna.append(soma)
    
for i in range(n):
    for j in range(n):
        aux = somaLinha[i] + somaColuna[j] - 2*valores[i][j]
        
        if resp < aux:
            resp = aux
            
print(resp)     
    

File: Problems/NepsAcademy/cpp/e45.cpp

#include <bits/stdc++.h>

using namespace std;

int main() {

    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, i, j, aux, resp = 0, soma = 0;

    cin >> n;

    int valores[n][n], somaLinha[n], somaColuna[n];


    for(i = 0; i < n; i++){
        soma = 0;
        for(j = 0; j < n; j++){
            cin >> aux;
            valores[i][j] = aux;
            soma += aux;
        }

        somaLinha[i] = soma; 
    }

    // Soma Coluna
    for(i = 0; i < n; i++){
        soma = 0;
        for(j = 0; j < n; j++){
            soma += valores[j][i];

        }
        somaColuna[i] = soma; 
    }

        
    // soma
    for(i = 0; i < n; i++){
        aux = 0;
        for(j = 0; j < n; j++){
            aux = somaLinha[i] + somaColuna[j] - 2 * valores[i][j];

            if (resp < aux){
                resp = aux;
            }
        }
    }


    cout << resp << "\n";


}